libs/eavmlib/src/network.erl

```diff
@@ -390,10 +370,25 @@ handle_info(Msg, State) ->
     {noreply, State}.
 
 %% @hidden
-terminate(_Reason, _State) ->
+%% Wait for port to be closed
+terminate(_Reason, State) ->
     Ref = make_ref(),
+    Pid = State#state.port,
+    PortMonitor = erlang:monitor(port, Pid),
     network_port ! {?SERVER, Ref, stop},
-    ok.
+    wait_for_port_close(PortMonitor, Pid).
+
+wait_for_port_close(PortMonitor, Pid) ->
+    receive
+        {'DOWN', PortMonitor, port, Pid, _DownReason} ->
+            ok;
+        _Other ->
+            % Handle unexpected messages if necessary
+            wait_for_port_close(PortMonitor, Pid)
+        % Timeout after 4000 milliseconds
+    after 4000 ->
```

Comment:
You might want to be able to pass the timeout as an option at startup and default to 4_000 if not provided.

Reply:
+1 on that idea.

Reply:
I digress, I don't see much value in this option at startup.

Things are in a bad state if you run into this timeout, almost to the point that it should call exit() and crash.

maybe have stop/1 that takes the timeout value? but still code/contracts for little/no purpose.

I fully expect this to in a supervisor and contracts be made for when it crashes - eg. if the network_port unexpectedly is gone you really want to crash and restart..

I'll call bugfix PR, and adding new startup options is beyond the scope;-)

Reply:
My thought was, as you pointed out, if it takes 4 seconds something has already gone terribly wrong and that is a long time to wait... if it were configurable I would set it to a few hundred milliseconds... but probably not worth adding that. 

Maybe we can just adjust the timeout, I would think a 2 second timeout would be more than sufficient here.

Reply:
I already changed it to 1 sec, yesterday, as 4 was a bit long;-)